### PR TITLE
Switch from strncpy to truncate_cpy

### PR DESCRIPTION
**Pull request by hrydgard**
Well, since strncpy has that ridiculous termination behavior, of course.

Suspect the one in sceKernelModule might be a crash from #9718 but not sure.

### CODE REVIEW

File: Core/HLE/sceIo.cpp [hunk 1]
```diff
@@ -2184,8 +2185,7 @@ static u32 sceIoDread(int id, u32 dirent_addr) {
 		PSPFileInfo &info = dir->listing[dir->index];
 		__IoGetStat(&entry->d_stat, info);
 
-		strncpy(entry->d_name, info.name.c_str(), 256);
-		entry->d_name[255] = '\0';
+		truncate_cpy(entry->d_name, info.name.c_str());
```
Comment (unknownbrackets) [inline Core/HLE/sceIo.cpp]: It's possible (though unlikely) that games may expect the zero-fill behavior of strncpy.

(and also for other PSP-visible structs - Module.)

-[Unknown]

Comment (hrydgard) [inline Core/HLE/sceIo.cpp]: Hm, yeah - for the PSP-internal structs we should probably stick to it. Fixing.

**Comment by hrydgard** [comment]
In the module case, the string is only sent along to our symbol map stuff - i think the truncation is fine (and even a fix).

**Comment by unknownbrackets** [comment]
Ah, missed that.

-[Unknown]

**Pull request merged**